File: YouTube/youtube/src/main/java/net/faithgen/youtube/YouTubeAdapter.java
package net.faithgen.youtube;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class YouTubeAdapter extends RecyclerView.Adapter<YouTubeHolder> {

    private Context context;
    private List<YouTubeObject> youTubeObjects;
    private LayoutInflater layoutInflater;

    public YouTubeAdapter(Context context, List<YouTubeObject> youTubeObjects) {
        this.context = context;
        this.youTubeObjects = youTubeObjects;
        this.layoutInflater = LayoutInflater.from(this.context);
    }

    @NonNull
    @Override
    public YouTubeHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new YouTubeHolder(layoutInflater.inflate(R.layout.list_youtube_item, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull YouTubeHolder holder, int position) {
        holder.getYouTubeView().setYouTubeObject(youTubeObjects.get(position));
    }

    @Override
    public int getItemCount() {
        if (youTubeObjects == null)
            return 0;
        return youTubeObjects.size();
    }
}

File: youtube/src/main/java/net/faithgen/youtube/YouTubeAdapter.kt
package net.faithgen.youtube

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

final class YouTubeAdapter(private val context: Context, private val youTubeObjects: List<YouTubeObject>?) : RecyclerView.Adapter<YouTubeHolder>() {
    private val layoutInflater: LayoutInflater by lazy { LayoutInflater.from(context) }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): YouTubeHolder {
        val youtubeView = layoutInflater.inflate(R.layout.list_youtube_item, parent, false) as YouTubeView
        return YouTubeHolder(youtubeView)
    }

    override fun getItemCount(): Int {
        if (youTubeObjects === null)
            return 0
        return youTubeObjects.size
    }

    override fun onBindViewHolder(holder: YouTubeHolder, position: Int) {
        holder.youTubeView.youTubeObject = youTubeObjects!![position]
    }
}